File: tests/unit/test_dados.py
"""Testes de `dados.py` (carregamento e filtros de jogos disputados/pendentes)."""

from __future__ import annotations

import pandas as pd
import pytest

from brasileirao_mc.dados import carregar_jogos, jogos_disputados, jogos_pendentes

CSV_EXEMPLO = """rodada,time_mandante,gols_mandante,time_visitante,gols_visitante
1,A,2,B,1
1,C,,D,
"""


@pytest.fixture
def csv_path(tmp_path):
    caminho = tmp_path / "jogos.csv"
    caminho.write_text(CSV_EXEMPLO)
    return str(caminho)


def test_carregar_jogos_le_todas_as_linhas(csv_path):
    jogos = carregar_jogos(csv_path)
    assert len(jogos) == 2


def test_carregar_jogos_valida_colunas(tmp_path):
    caminho = tmp_path / "invalido.csv"
    caminho.write_text("rodada,time_mandante\n1,A\n")

    with pytest.raises(ValueError):
        carregar_jogos(str(caminho))


def test_jogos_disputados_e_pendentes(csv_path):
    jogos = carregar_jogos(csv_path)

    assert len(jogos_disputados(jogos)) == 1
    assert len(jogos_pendentes(jogos)) == 1
    assert jogos_pendentes(jogos).iloc[0]["time_mandante"] == "C"

File: tests/testthat/test-dados.R
# Testes de `dados.R` (carregamento e filtros de jogos disputados/pendentes).

.csv_exemplo <- c(
  "rodada,time_mandante,gols_mandante,time_visitante,gols_visitante",
  "1,A,2,B,1",
  "1,C,,D,"
)

.criar_csv_temp <- function(linhas) {
  caminho <- tempfile(fileext = ".csv")
  writeLines(linhas, caminho)
  caminho
}

test_that("carregar_jogos le todas as linhas", {
  caminho <- .criar_csv_temp(.csv_exemplo)

  jogos <- carregar_jogos(caminho)
  expect_equal(nrow(jogos), 2)
})

test_that("carregar_jogos valida colunas", {
  caminho <- .criar_csv_temp(c("rodada,time_mandante", "1,A"))

  expect_error(carregar_jogos(caminho))
})

test_that("jogos_disputados e jogos_pendentes filtram corretamente", {
  caminho <- .criar_csv_temp(.csv_exemplo)
  jogos <- carregar_jogos(caminho)

  expect_equal(nrow(jogos_disputados(jogos)), 1)
  expect_equal(nrow(jogos_pendentes(jogos)), 1)
  expect_equal(jogos_pendentes(jogos)$time_mandante[1], "C")
})
